test/parallel/test-strace-openat-openssl.js

```diff
@@ -0,0 +1,50 @@
+'use strict';
+
+const common = require('../common');
+const { spawn, spawnSync } = require('node:child_process');
+const { createInterface } = require('node:readline');
+const assert = require('node:assert');
+
+if (!common.hasCrypto)
+  common.skip('missing crypto');
+if (!common.isLinux)
+  common.skip('linux only');
+if (spawnSync('strace').error !== undefined) {
+  common.skip('missing strace');
+}
+
+{
+  const allowedOpenCalls = new Set([
+    '/etc/ssl/openssl.cnf',
+  ]);
+  const strace = spawn('strace', [
+    '-f', '-ff',
+    '-e', 'trace=openat',
+    '-s', '128',
+    '-D', process.execPath, '-e', 'require("crypto")',
+  ]);
+
+  // stderr is the default for strace
+  const rl = createInterface({ input: strace.stderr });
+  rl.on('line', (line) => {
+    if (!line.startsWith('openat')) {
+      return;
+    }
+
+    const file = line.match(/"(.*?)"/)[1];
+    // skip .so reading attempt
+    if (file.match(/.+\.so(\.?)/) !== null) {
```

Comment:
It might actually not be that bad to assert these as well – if we added a new .so dependency on Linux, we might want to know about that?

Reply:
In my case, it's reading the `.so` from shared locations, for instance: `"/home/rafaelgss/.gvm/pkgsets/go1.15/global/overlay/lib/libpthread.so.0"`. How would we handle it?

Reply:
You could assert just the filename, ignoring the rest of the path, right?